sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCV2Suite.scala

```diff
@@ -0,0 +1,210 @@
+import org.apache.spark.util.Utils
+
+class JDBCV2Suite extends QueryTest with SharedSparkSession {
+  import testImplicits._
+
+  val tempDir = Utils.createTempDir()
+  val url = s"jdbc:h2:${tempDir.getCanonicalPath};user=testUser;password=testPass"
+  var conn: java.sql.Connection = null
+
+  override def sparkConf: SparkConf = super.sparkConf
+    .set("spark.sql.catalog.h2", classOf[JDBCTableCatalog].getName)
+    .set("spark.sql.catalog.h2.url", url)
+    .set("spark.sql.catalog.h2.driver", "org.h2.Driver")
+
+  private def withConnection[T](f: Connection => T): T = {
+    val conn = DriverManager.getConnection(url, new Properties())
+    try {
+      f(conn)
+    } finally {
+      conn.close()
+    }
+  }
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    Utils.classForName("org.h2.Driver")
+    withConnection { conn =>
+      conn.prepareStatement("CREATE SCHEMA \"test\"").executeUpdate()
+      conn.prepareStatement(
+        "CREATE TABLE \"test\".\"empty_table\" (name TEXT(32) NOT NULL, id INTEGER NOT NULL)")
+        .executeUpdate()
+      conn.prepareStatement(
+        "CREATE TABLE \"test\".\"people\" (name TEXT(32) NOT NULL, id INTEGER NOT NULL)")
+        .executeUpdate()
+      conn.prepareStatement("INSERT INTO \"test\".\"people\" VALUES ('fred', 1)").executeUpdate()
+      conn.prepareStatement("INSERT INTO \"test\".\"people\" VALUES ('mary', 2)").executeUpdate()
+    }
+  }
+
+  override def afterAll(): Unit = {
+    Utils.deleteRecursively(tempDir)
+    super.afterAll()
+  }
+
+  test("simple scan") {
+    checkAnswer(sql("SELECT * FROM h2.test.empty_table"), Seq())
+    checkAnswer(sql("SELECT * FROM h2.test.people"), Seq(Row("fred", 1), Row("mary", 2)))
+    checkAnswer(sql("SELECT name, id FROM h2.test.people"), Seq(Row("fred", 1), Row("mary", 2)))
+  }
+
+  test("scan with filter push-down") {
+    val df = spark.table("h2.test.people").filter($"id" > 1)
+    val filters = df.queryExecution.optimizedPlan.collect {
+      case f: Filter => f
+    }
+    assert(filters.isEmpty)
+    checkAnswer(df, Row("mary", 2))
+  }
+
+  test("scan with column pruning") {
+    val df = spark.table("h2.test.people").select("id")
+    val scan = df.queryExecution.optimizedPlan.collectFirst {
+      case s: DataSourceV2ScanRelation => s
+    }.get
+    assert(scan.schema.names.sameElements(Seq("ID")))
+    checkAnswer(df, Seq(Row(1), Row(2)))
+  }
+
+  test("scan with filter push-down and column pruning") {
+    val df = spark.table("h2.test.people").filter($"id" > 1).select("name")
+    val filters = df.queryExecution.optimizedPlan.collect {
+      case f: Filter => f
+    }
+    assert(filters.isEmpty)
+    val scan = df.queryExecution.optimizedPlan.collectFirst {
+      case s: DataSourceV2ScanRelation => s
+    }.get
+    assert(scan.schema.names.sameElements(Seq("NAME")))
+    checkAnswer(df, Row("mary"))
+  }
+
+  // TODO (SPARK-32592): this doesn't work because `DataFrameReader.table` ignores the options.
```

Comment:
can we enable the test now? the bug is fixed.

Reply:
@cloud-fan I enabled other tests. I will enable this one once the other PR is merged, otherwise the assertion `assert(df.rdd.getNumPartitions == 2)` will fail.